crates/rspack_core/src/utils/visitor.rs

```diff
@@ -62,9 +62,21 @@ impl ExpressionInfoCallExpression {
   }
 }
 
+#[derive(Debug)]
+pub struct ExpressionInfoMemberExpression {
+  object: MemberExpr,
+}
+
+impl ExpressionInfoMemberExpression {
+  pub fn obj(&self) -> &MemberExpr {
+    &self.object
+  }
+}
+
 #[derive(Debug)]
 pub enum ExpressionInfoKind {
   CallExpression(Box<ExpressionInfoCallExpression>),
+  MemberExpression(Box<ExpressionInfoMemberExpression>),
```

Comment:
I don't think this is the optimal solution. We performed many similar tasks during the ﻿`walk_member`, and our ultimate goal is to align with webpack.

Reply:
This is aligning with webpack. Webpack walk member expression and if exports[xxx].aaa, it will return object with member expression, then the return value of getMemberExpressionInfo  will be undefined, and the assignMembers hook will not be triggered.

```js
extractMemberExpressionChain(expression) {
	/** @type {AnyNode} */
	let expr = expression;
	const members = [];
	const membersOptionals = [];
	const memberRanges = [];
	// exports[xxx]
	while (expr.type === "MemberExpression") {
		if (expr.computed) { // is computed
			if (expr.property.type !== "Literal") break; // is not literal, break and object = expr
			members.push(`${expr.property.value}`); // the literal
			memberRanges.push(expr.object.range); // the range of the expression fragment before the literal
		}
                // ...
	}

	return {
		members,
		membersOptionals,
		memberRanges,
		object: expr // object = exports[xxx] and it is a member expression
	};
}

getMemberExpressionInfo(expression, allowedTypes) {
	const { object, members, membersOptionals, memberRanges } =
		this.extractMemberExpressionChain(expression);
	// object = exports[xxx], object.type === MemberExpression
	// return undefined
	switch (object.type) {
		case "CallExpression": {
			// ...
		}
		case "Identifier":
		case "MetaProperty":
		case "ThisExpression": {
			// ...
		}
	}
}
```

Reply:
No, what I mean is we've already done the same thing in [javascript_parser::MemberExpressionInfo](https://github.com/web-infra-dev/rspack/blob/main/crates/rspack_plugin_javascript/src/visitors/dependency/parser/mod.rs#L56-L60). Therefore, this change seems unnecessary.

Furthermore, I think that ﻿`utils::extract_member_expression_chain` can ultimately be removed.

Reply:
Here needs to distinguish between `exports[xxx]` and `exports`, currently the `utils::extract_member_expression_chain` will return an empty array in both cases, and then make the logic of handling `exports` ran on `exports[xxx]`. Is there any other recommended way?

Reply:
> Furthermore, I think that ﻿`utils::extract_member_expression_chain` can ultimately be removed.

I agree

Reply:
> Is there any other recommended way?

If I understand correctly, I believe we need to port over this logic: https://github.com/webpack/webpack/blob/main/lib/dependencies/CommonJsExportsParserPlugin.js#L308-L349.

Reply:
> Perhaps I can use parser.get_member_expression_info to handle this?

No method to get members on parser::MemberExpressionInfo. T_T

Reply:
> Perhaps I can use parser.get_member_expression_info to handle this?

If it functions as anticipated, I think we can consider this as a temporary solution.